File: 2.Dagger/ComponentTypes/CustomFactory/src/main/java/com/app/customfactory/MainActivity.java
package com.app.customfactory;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;

import javax.inject.Inject;

public class MainActivity extends AppCompatActivity {
    AppComponent appComponent;

    @Inject
    String firstString;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        /**
         * В данном случае мы сразу увидем ошибку если не передали нужные параметры
         * */

        appComponent = DaggerAppComponent
                .factory()
                .create(9,true);

        appComponent.inject(this);

        Log.i(TAG, "onCreate: "+firstString);
    }

    private static final String TAG = "MainActivity";
}

File: 3.Koin/PassingArguments/app/src/main/java/com/app/passingarguments/MainActivity.kt
package com.app.passingarguments

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import org.koin.android.ext.android.inject
import org.koin.android.ext.koin.androidLogger
import org.koin.core.context.startKoin

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        startKoin {
            androidLogger()
            modules(dataModule)
        }
        val a:A by inject()
        Log.i(TAG, a.number.toString())
    }

    companion object {
        private const val TAG = "MainActivity"
    }
}